### PR TITLE
document set_default_verify_paths caveats

### LINKED ISSUE #642: Document set_default_verify_paths caveats for macOS/Windows users
Users with the statically linked cryptography wheel have some potentially confusing behaviors when they call `set_default_verify_paths`. Let's document that:

* macOS will work, but only if the user has the openssl@1.1 formula installed (in the default location) from homebrew
* windows will not work.
* manylinux1 will work using #633

### PR DESCRIPTION
**Pull request by reaperhulk**
fixes #642

### CODE REVIEW

**Comment by tiran** [comment]
How did you get manylinux1 binaries working? Does pyopenssl 17.1.0 detect distribution and look up paths?

**Comment by alex** [comment]
Pretty much, yes.

**Comment by reaperhulk** [comment]
@tiran we use a sentinel value so pyOpenSSL can easily detect if it's running under manylinux1. If and only if that occurs does it use the fallback paths to find the root stores.

**Comment by tiran** [comment]
Could you please add another sentence that mentions this behavior?

**Comment by reaperhulk** [comment]
@tiran How's that?

File: doc/api/ssl.rst [hunk 1]
```diff
@@ -307,8 +307,15 @@ Context objects have the following methods:
...
     Specify that the platform provided CA certificates are to be used for
-    verification purposes.  This method may not work properly on OS X.
-
+    verification purposes. This method has some caveats related to the
+    binary wheels that cryptography (pyOpenSSL's primary dependency) ships:
+
+    * macOS will only load certificates using this method if the user has
+      the ``openssl@1.1`` Homebrew formula installed in the default location.
+    * Windows will not work.
+    * manylinux1 cryptography wheels will work in pyOpenSSL 17.1.0 and above.
```
Comment (alex) [inline doc/api/ssl.rst]: Maybe we should be clearer "will work on most common linux distributions" or something

Comment (reaperhulk) [inline doc/api/ssl.rst]: Part of me wants to say that prevarication in the absence of an actual example is annoying, but undoubtedly somewhere somehow somebody is going to find a way to build a linux distro that won't work, so okay.

**Pull request merged**